online-store/src/components/filter/filter.ts

```diff
@@ -0,0 +1,572 @@
+import './filter.scss';
+import { filterSelector, productsArrayI, sliderSelector } from '../types';
+import { productsData, CreateProductCard } from '../cards/cards';
+import { SortProductCard } from '../sort/sort';
+import { slider } from './slider/slider';
+import { searchProductCard } from '../search/search';
+import { searchParams } from '../routing/routing';
+
+const filterCategory: filterSelector = { arrFilter: [], countFilter: [] }; // категории по всему списку товаров
+const filterBrand: filterSelector = { arrFilter: [], countFilter: [] }; // брэнды по всему списку товаров
+const filterStock: sliderSelector = { min: 0, max: 1 }; // минимальное и максимальное кол-во товара по всему списку товаров
+const filterPrice: sliderSelector = { min: 0, max: 1 }; // минимальная и максимальная стоимость товара по всему списку товаров
+
+function CreateFilter(
```

Comment:
this is probably a very large function that does a lot of things. It is worth avoiding this. Large functions are very complex in the future